peer.go

```diff
@@ -784,11 +806,34 @@ out:
 			peerLog.Infof("New channel active ChannelPoint(%v) "+
 				"with peerId(%v)", chanPoint, p.id)
 
-			decoder := htlcswitch.NewSphinxDecoder(p.server.sphinx)
 			link := htlcswitch.NewChannelLink(
 				htlcswitch.ChannelLinkConfig{
-					Peer:             p,
-					DecodeOnion:      decoder.Decode,
+					Peer:                  p,
+					DecodeHopIterator:     p.server.sphinx.DecodeHopIterator,
+					DecodeOnionObfuscator: p.server.sphinx.DecodeOnionObfuscator,
+					ChannelEdgeInfo: func(chanID lnwire.ChannelID) (*channeldb.ChannelEdgePolicy,
```

Comment:
Similar comment here. 

Also as this logic is repeated above, a function should be added which factors out the logic. Note that the function need not have a struct as its target (a method), it can be free form.